exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -386,6 +380,42 @@ def as_otlp_any_value(value)
           result
         end
 
+        # Returns the url used by the exporter based on the precedence defined in the specification:
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp
+        #
+        # @param [String, nil] endpoint
+        # @return [URI::Generic]
+        def determine_url(endpoint)
+          if !endpoint.nil?
+            # Use endpoint provided via argument as-is
+            parse_url_or_raise(endpoint)
+          elsif !ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT'].nil?
+            # Use signal-specific endpoint as-is
+            parse_url_or_raise(ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT'])
+          elsif !ENV['OTEL_EXPORTER_OTLP_ENDPOINT'].nil?
+            # Append v1/traces only if the non-signal specific env var is set.
+            non_signal_specific_endpoint = ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
+            # append '/' if not present
+            non_signal_specific_endpoint += '/' unless non_signal_specific_endpoint.end_with?('/')
```

Comment:
This is not necessary. The `URI.join` call below will insert the `/` if required.

Reply:
Hmm, I could be wrong (this is the first time I'm using ruby). I also initially assumed it does, but trying it out in `irb`:

```
$ irb
3.2.0 :001 > require('uri')
 => true 
3.2.0 :002 > URI.join('http://test.com/api','v2/url')
 => #<URI::HTTP http://test.com/v2/url> 
```

drops `api` from the path. I think it may be debatable if it is directly intended by the [spec here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp) to append it. However, I think many language implementations do append '/' if it's not there (examples: [JavaScript](https://github.com/open-telemetry/opentelemetry-js/blob/abfe059d72d2a50d993ee7fb5ea50a1c29851a31/experimental/packages/otlp-exporter-base/src/util.ts#L52), [Python](https://github.com/open-telemetry/opentelemetry-python/blob/c9277ffb137fad999ee14bee4f99bad2d00b8b03/exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py#L247), [Java](https://github.com/open-telemetry/opentelemetry-java/blob/ea7b761757caae434825a45ee6e7d9e559fe76cd/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java#L78))